pkg/cluster/fixetcd.go

```diff
@@ -0,0 +1,517 @@
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: serviceAccountName,
+					Containers: []corev1.Container{
+						{
+							Name:  "backup",
+							Image: image,
+							Command: []string{
+								"/bin/bash",
+								"-cx",
+								backupOrFixEtcd,
+							},
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: to.BoolPtr(true),
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "PEER_PODS",
+									Value: peerPods,
+								},
+								{
+									Name:  "DEGRADED_NODE",
+									Value: de.Node,
+								},
+								{
+									Name:  "FIX_PEERS",
+									Value: "true",
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "host",
+									MountPath: "/host",
+									ReadOnly:  false,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{
+		FieldManager: jobNameFixPeers,
+	})
+	if err != nil {
+		return err
+	}
+
+	timeout := adjustTimeout(ctx, log)
+	log.Infof("Waiting for %s", jobNameFixPeers)
+	select {
+	case <-time.After(timeout):
+		log.Infof("Finished waiting for job %s", jobNameFixPeers)
+		break
+	case <-ctx.Done():
+		log.Warnf("Request cancelled while waiting for %s", jobNameFixPeers)
+	}
+
+	log.Infof("Deleting %s now", jobNameFixPeers)
+	propPolicy := metav1.DeletePropagationBackground
+	err = kubecli.BatchV1().Jobs(b.Namespace).Delete(ctx, jobNameFixPeers, metav1.DeleteOptions{
+		PropagationPolicy: &propPolicy,
+	})
+	if err != nil {
+		return err
+	}
+
+	// return errors from deferred cleanup function
+	return err
+}
+
+func createPrivilegedServiceAccount(ctx context.Context, log *logrus.Entry, name, usersAccount string, kubecli kubernetes.Interface, securitycli securityv1client.SecurityContextConstraintsInterface) (func() error, error) {
+	log.Infof("Creating service account %s now", name)
+	serviceAcc, err := kubecli.CoreV1().ServiceAccounts(nameSpaceEtcds).Create(ctx, &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		AutomountServiceAccountToken: to.BoolPtr(true),
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Creating cluster role %s now", usersAccount)
+	cr, err := kubecli.RbacV1().ClusterRoles().Create(ctx, &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: usersAccount,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:     []string{"get", "create"},
+				Resources: []string{"pods", "pods/exec"},
+				APIGroups: []string{""},
+			},
+		},
+	}, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Creating cluster role binding %s now", serviceAccountName)
```

Comment:
Would it make sense to replace `serviceAccountName` with `name` here and throughout this function? I got a little confused by the fact that the same underlying constant value is being both directly and through a function parameter.